runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/MemoryMonitor.java

```diff
@@ -467,6 +474,14 @@ private void shutDownDueToGcThrashing(int thrashingCount) {
   public void run() {
     synchronized (waitingForStateChange) {
       Preconditions.checkState(!isRunning.getAndSet(true), "already running");
+
+      if (this.GCThrashingPercentagePerPeriod <= 0 || this.GCThrashingPercentagePerPeriod >= 100) {
+        LOG.warn(
+            "GCThrashingPercentagePerPeriod: {} is not valid value. Not starting MemoryMonitor.",
```

Comment:
Nit: this sounds a bit like a warning of user error, when actually it might be what they want. Ok to leave as-is.